src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs

```diff
@@ -126,24 +123,19 @@ private void WarmupTransactions(ParallelOptions parallelOptions, IReleaseSpec sp
 
         try
         {
-            int progress = 0;
-            Parallel.For(0, block.Transactions.Length, parallelOptions, _ =>
+            ParallelUnbalancedWork.For(0, block.Transactions.Length, parallelOptions, (preWarmer: this, block, stateRoot, spec), static (i, state) =>
```

Comment:
for readability create a private struct for `State` with optional `spec` and use it in this class.